### PR TITLE
SFT patch: (1) enable sequence parallelism and (2) enable profile

### PR DESCRIPTION
**Pull request by erhoo82**
# What does this PR do ?

SFT patch: (1) enable sequence parallelism and (2) enable profile

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py [hunk 1]
```diff
@@ -94,8 +94,12 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
         else:
             base_module = self.model
 
+        # Set the profile start and end steps in the unit of global batach
+        if hasattr(self, '_nsys_profile_enabled'):
+            self._nsys_profile_start_step = self.cfg.nsys_profile.get('start_step', 0)
+            self._nsys_profile_end_step = self.cfg.nsys_profile.get('end_step', 0)
+
         self._reset_activation_checkpointing_args()
-        self._reset_sequence_parallelism_args()
```
Comment (ericharper) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py:98]: This is good we need to remove from the constructor.

**Comment by ericharper** [comment]
jenkins

**Review by hsiehjackson — COMMENTED** [review]
@erhoo82 one question: if we want to do inference, do we still need this [assert](https://github.com/NVIDIA/NeMo/blob/2ecf71a35d9e2bf48cb300e19d661332905046e5/nemo/collections/nlp/modules/common/text_generation_utils.py#L731-L734) in generation mode?

**Comment by hsiehjackson** [comment]
@erhoo82 for inference do you mean generation mode instead of calculating validation loss? This assert needs config `sequence_parallel == False`. I think if you remove `_reset_sequence_parallelism_args` for  models with sequence parallel, we may get assertion error during generation.

**Review by hsiehjackson — APPROVED** [review]
LGTM

**Comment by hsiehjackson** [comment]
jenkins

**Pull request merged**